iothub_client/src/iothub_client_core.c

```diff
@@ -1734,11 +1735,10 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_SetOption(IOTHUB_CLIENT_CORE_HANDLE iotHub
             /* Codes_SRS_IOTHUBCLIENT_41_005: [ If parameter `optionName` is `OPTION_MESSAGE_TIMEOUT` then `IoTHubClientCore_SetOption` shall set `currentMessageTimeout` parameter of `IoTHubClientInstance` ]*/
             else if (strcmp(OPTION_MESSAGE_TIMEOUT, optionName) == 0)
             {
-                tickcounter_ms_t foo = (tickcounter_ms_t)(*(tickcounter_ms_t*)value);
-                iotHubClientInstance->currentMessageTimeout = foo;
+                iotHubClientInstance->currentMessageTimeout = (tickcounter_ms_t)(*(tickcounter_ms_t*)value);
 
```

Comment:
The extra (tickcounter_ms_t) cast is extra. You do the same on line 1720. 
Also, nit: please add a space in your pointer casts to be consistent. Currently there are few that are type* and some that are type *

Reply:
updated to reflect this